plugins/android-boot/fu-android-boot-device.c

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright (C) 2022 Dylan Van Assche <me@dylanvanassche.be>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <fcntl.h>
+#include <string.h>
+
+#include "fu-android-boot-device.h"
+
+#define ANDROID_BOOT_PROC_CMDLINE     "/proc/cmdline"
+#define ANDROID_BOOT_MAX_CMDLINE_SIZE 4096
+#define ANDROID_BOOT_BOOT_SLOT_ARG    "androidboot.slot_suffix"
+#define ANDROID_BOOT_SERIAL_ARG	      "androidboot.serialno"
+#define ANDROID_BOOT_UNKNOWN_VERSION  "0.0.0"
+#define ANDROID_BOOT_SECTOR_SIZE      512
+
+struct _FuAndroidBootDevice {
+	FuUdevDevice parent_instance;
+	const gchar *label;
+	const gchar *boot_slot;
+	const gchar *uuid;
+	const gchar *version;
+	guint64 max_size;
+};
+
+G_DEFINE_TYPE(FuAndroidBootDevice, fu_android_boot_device, FU_TYPE_UDEV_DEVICE)
+
+static void
+fu_android_boot_device_to_string(FuDevice *device, guint idt, GString *str)
+{
+	FuAndroidBootDevice *self = FU_ANDROID_BOOT_DEVICE(device);
+
+	fu_string_append(str, idt, "BootSlot", self->boot_slot);
+	fu_string_append(str, idt, "Label", self->label);
+	fu_string_append(str, idt, "UUID", self->uuid);
+	fu_string_append_kx(str, idt, "MaxSize", self->max_size);
+}
+
+static gboolean
+fu_android_boot_device_probe(FuDevice *device, GError **error)
```

Comment:
In this function you use a lot of parsing gudev attributes directly.  I think most of this should move to fu-udev-device.c for this device type.

Reply:
I'm not so worried; if there was two plugins reading the same values perhaps, but for one I don't mind so much.